src/main/java/sirius/biz/tycho/kb/KnowledgeBase.java

```diff
@@ -247,18 +247,20 @@ public List<KnowledgeBaseArticle> query(String language, String query, int maxRe
 
         List<KnowledgeBaseArticle> result = new ArrayList<>();
         elastic.select(KnowledgeBaseEntry.class)
-               .eq(KnowledgeBaseEntry.LANG, language)
-               .queryString(query, QueryField.contains(KnowledgeBaseEntry.SEARCH_FIELD))
+               .eqIgnoreNull(KnowledgeBaseEntry.LANG, language)
+               .queryString(query,
+                            QueryField.contains(KnowledgeBaseEntry.SEARCH_FIELD))
                .limit(maxResults * 5)
                .iterateAll(entry -> {
                    if (entry.checkPermissions()) {
                        result.add(new KnowledgeBaseArticle(entry, language, this));
                    }
                });
```

Comment:
Optional: extract Method for query and permission check which calls consumer if entry can be accessed (could replace both very similar queries)